simcase-game/src/views/Ranking.js

```diff
@@ -69,13 +65,165 @@ const button = {
+            points: doc.data().points
+          });
+        });
+        data.push({ username: "You", points: score });
+        data.sort((a, b) => {
+          return b.points - a.points;
+        });
+        setData(data);
+        data.filter((value, index) => {
+          if (value.points === score) {
+            setPosition(index + 1);
+          }
+        });
+      });
+  }
+
+  function addUser() {
+    firebase
+      .firestore()
+      .collection("data")
+      .add({
+        username: user,
+        points: score
+      });
+
+    fire();
+    setScoreboard(false);
+  }
+
+  function handleChange(e) {
+    setUser(e.target.value);
+  }
+
   return (
     <>
       <Navbar />
       <MainContent styles={container}>
         <h2>Leaderboard</h2>
+        {scoreboard && (
+          <>
+            <div
+              css={{
+                display: "flex",
+                background:
+                  "linear-gradient(white, white) no-repeat center/2px 100%",
+                justifyContent: "space-between",
+                width: "150px",
+                margin: "0 auto",
+                padding: "20px",
+                backgroundColor: "#0C4785",
+                color: "white",
+                borderRadius: "10px"
+              }}
+            >
+              <div css={{ display: "flex", flexDirection: "column" }}>
+                <span css={{ fontSize: ".7em" }}>RANK</span>
+                <strong
+                  css={{
+                    padding: "0 15px",
+                    marginTop: "10px"
+                  }}
+                >
+                  {position}
+                </strong>
+              </div>
+              <div css={{ display: "flex", flexDirection: "column" }}>
+                <span css={{ fontSize: ".7em" }}>SCORE</span>
+                <strong
+                  css={{
+                    padding: "0 15px",
+                    marginTop: "10px"
+                  }}
+                >
+                  {score}
+                </strong>
+              </div>
+            </div>
+            <form onSubmit={addUser} css={{ margin: "20px 0" }}>
+              <div
+                css={{
+                  display: "flex",
+                  flexDirection: "column",
+                  width: "250px",
+                  margin: "10px auto",
+                  backgroundColor: "#0C4785",
+                  padding: "10px",
+                  color: "white",
+                  borderRadius: "10px"
+                }}
+              >
+                <label
+                  css={{
+                    fontSize: ".85em",
+                    marginBottom: "20px"
+                  }}
+                >
+                  NAME
+                </label>
+                <input
+                  type="text"
+                  name="user"
+                  onChange={handleChange}
+                  css={{
+                    padding: "7px",
```

Comment:
What do you think about creating a const and use instead of this lines of CSS inside the component?

Reply:
Yes, I forgot to use the way you are styles css. I am going to refactor this.